### PR TITLE
AArch64: Improve mFirstTrue and mLastTrue evaluators

### PR DESCRIPTION
**Pull request by Akira1Saitoh**
This commit improves instruction sequence generated in mFirstTrue and mLastTrue evaluators.

Signed-off-by: Akira Saitoh <saiaki@jp.ibm.com>

### CODE REVIEW

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -1335,81 +1335,58 @@ OMR::ARM64::TreeEvaluator::mFirstTrueEvaluator(TR::Node *node, TR::CodeGenerator
...
-          * bfi     x0, x0, #24, #20            ; Inserts bit 0 - 19 into bit 24 - 43.
-          * ubfx    x0, x0, #36, #16            ; Moves bit 36 - 51 to bit 0 - 15.
-          * orr     w0, w0, #0x10000            ; Sets the bit 16.
-          * rbit    w0, w0                      ; Reverses bits.
-          * clz     w0, w0                      ; Counts leading zeros.
+          * shrn    v1.8b, v0.8h, #4            ; Moves mask values to bit 0-3 and 4-7 of the lane 0 - 7.
+          * umov    x0, v1.2d[0]
+          * rbit    x0, x0                      ; Reverses bits.
+          * clz     x0, x0                      ; Counts leading zeros.
+          * lsr     x0, x0, #2                  ; Divide by 4.
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRTreeEvaluator.cpp:1342]: "Divide" -> "Divides"

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp [hunk 2]
```diff
@@ -1335,81 +1335,58 @@ OMR::ARM64::TreeEvaluator::mFirstTrueEvaluator(TR::Node *node, TR::CodeGenerator
...
+          * shrn    v1.4h, v0.4s, #8            ; Moves mask values to byte 0 - 7.
           * umov    w0, v1.2d[0]
-          * bfi     x0, x0, #28, #18            ; Inserts bit 0 - 17 into bit 28 - 45.
-          * ubfx    x0, x0, #42, #8             ; Moves bit 42 - 49 to bit 0 - 7.
-          * orr     w0, w0, #0x100              ; Sets the bit 8.
-          * rbit    w0, w0                      ; Reverses bits.
-          * clz     w0, w0                      ; Counts leading zeros.
+          * rbit    x0, x0                      ; Reverses bits.
+          * clz     x0, x0                      ; Counts leading zeros.
+          * lsr     x0, x0, #3                  ; Divide by 8.
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRTreeEvaluator.cpp:1356]: The same comment as above.

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp (hunk 1 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRTreeEvaluator.cpp]: fixed.

File: compiler/aarch64/codegen/OMRTreeEvaluator.cpp (hunk 2 above)
Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRTreeEvaluator.cpp]: fixed.

**Comment by knn-k** [comment]
Jenkins build aarch64,amac

**Review by knn-k — APPROVED** [review]

**Comment by knn-k** [comment]
amac build is unavailable today.  Build on AArch64 Linux was successful.  Merging.

**Pull request merged**